apidef/oas/security.go

```diff
@@ -95,19 +95,48 @@ func (s *OAS) extractTokenTo(api *apidef.APIDefinition, name string) {
 
 // JWT holds the configuration for the JWT middleware.
 type JWT struct {
-	Enabled                 bool `bson:"enabled" json:"enabled"` // required
-	AuthSources             `bson:",inline" json:",inline"`
-	Source                  string   `bson:"source,omitempty" json:"source,omitempty"`
-	SigningMethod           string   `bson:"signingMethod,omitempty" json:"signingMethod,omitempty"`
-	IdentityBaseField       string   `bson:"identityBaseField,omitempty" json:"identityBaseField,omitempty"`
-	SkipKid                 bool     `bson:"skipKid,omitempty" json:"skipKid,omitempty"`
-	PolicyFieldName         string   `bson:"policyFieldName,omitempty" json:"policyFieldName,omitempty"`
-	ClientBaseField         string   `bson:"clientBaseField,omitempty" json:"clientBaseField,omitempty"`
-	Scopes                  *Scopes  `bson:"scopes,omitempty" json:"scopes,omitempty"`
-	DefaultPolicies         []string `bson:"defaultPolicies,omitempty" json:"defaultPolicies,omitempty"`
-	IssuedAtValidationSkew  uint64   `bson:"issuedAtValidationSkew,omitempty" json:"issuedAtValidationSkew,omitempty"`
-	NotBeforeValidationSkew uint64   `bson:"notBeforeValidationSkew,omitempty" json:"notBeforeValidationSkew,omitempty"`
-	ExpiresAtValidationSkew uint64   `bson:"expiresAtValidationSkew,omitempty" json:"expiresAtValidationSkew,omitempty"`
+	// Enabled activates the basic authentication mode.
+	Enabled bool `bson:"enabled" json:"enabled"` // required
+
+	// AuthSources configures the source for the JWT.
+	AuthSources `bson:",inline" json:",inline"`
+
+	// Source contains the source for the JWT.
+	Source string `bson:"source,omitempty" json:"source,omitempty"`
+
+	// SigningMethod contains the signing method to use for the JWT.
+	SigningMethod string `bson:"signingMethod,omitempty" json:"signingMethod,omitempty"`
+
+	// IdentityBaseField specifies the claim name uniquely identifying the subject of the JWT.
+	// The identity fields that are checked in order are: `kid`, IdentityBaseField, `sub`.
+	IdentityBaseField string `bson:"identityBaseField,omitempty" json:"identityBaseField,omitempty"`
+
+	// SkipKid controls skipping using the `kid` claim from a JWT (default behaviour).
+	// When this is true, the field configured in IdentityBaseField is checked first.
+	SkipKid bool `bson:"skipKid,omitempty" json:"skipKid,omitempty"`
+
+	// PolicyFieldName is a configurable claim name from which a policy ID is extracted.
+	// The policy is applied to the session as a base policy.
+	PolicyFieldName string `bson:"policyFieldName,omitempty" json:"policyFieldName,omitempty"`
+
+	// Deprecated: ClientBaseField is unused.
```

Comment:
What should be used instead of ClientBaseField?

Reply:
Nothing, can't say what it was intended for